app/src/main/java/protect/card_locker/preferences/myAppCompatActivity.java

```diff
@@ -0,0 +1,58 @@
+package protect.card_locker.preferences;
+
+import android.content.SharedPreferences;
+import android.content.res.Resources;
+import android.widget.Toast;
+
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
+
+import protect.card_locker.R;
+
+public class myAppCompatActivity extends AppCompatActivity {
+
+    SharedPreferences pref;
+//
+        @Override
+    public Resources.Theme getTheme() {
+        Resources.Theme theme = super.getTheme();
+            pref = PreferenceManager
+                    .getDefaultSharedPreferences(getApplicationContext());
+            String themeName = pref.getString("pref_theme_color", "App theme");
+            switch (themeName){
+                case "Pink theme":
```

Comment:
Please match using the settings_key_pink_theme directly, so that this won't break if the settings keys change for any reason (they shouldn't, but I just don't want this to break if it does for some weird reason).

Reply:
static final String can be used in the switch case and I am not getting any way to get string.xml value as static.
maybe I can implement in if-else.

Reply:
Ah...

Maybe something like:
```
public class [...] {
    Map<Integer, Integer> themes = new HashMap<Integer, Integer>();

    public Resource.Theme getTheme() {
        [...]
        if (themes.isEmpty()) {
            themes.put(R.string.settings_key_blue_theme, R.style.AppTheme_blue);
            [...]
        }

        applyTheme(themes.get(theme));
```

But maybe if/else will be simpler and better, not sure...

Reply:
I implemented like....

`if(themeName.equals(getString(R.string.settings_key_catima_theme))){
                theme.applyStyle(R.style.AppTheme_NoActionBar, true);
            }else if(themeName.equals(getString(R.string.settings_key_pink_theme))){
                theme.applyStyle(R.style.AppTheme_pink, true);
            }else if(themeName.equals(getString(R.string.settings_key_magenta_theme))){
                theme.applyStyle(R.style.AppTheme_magenta, true);
            }else if(themeName.equals(getString(R.string.settings_key_violet_theme))){
                theme.applyStyle(R.style.AppTheme_violet, true);
            }else if(themeName.equals(getString(R.string.settings_key_blue_theme))){
                theme.applyStyle(R.style.AppTheme_blue, true);
            }else if(themeName.equals(getString(R.string.settings_key_sky_blue_theme))){
                theme.applyStyle(R.style.AppTheme_sky_blue, true);
            }else if(themeName.equals(getString(R.string.settings_key_green_theme))){
                theme.applyStyle(R.style.AppTheme_green, true);
            }else if(themeName.equals(getString(R.string.settings_key_grey_theme))){
                theme.applyStyle(R.style.AppTheme_grey, true);
            }else {
                theme.applyStyle(R.style.AppTheme_brown, true);
            }`

is it good?

Reply:
Yeah that is fine, but please make the last else use the Catima default theme instead of brown. If a new theme is added and someone forgets to add it here, I want the app to use the Catima theme instead of brown theme.